src/conference.go

```diff
@@ -0,0 +1,226 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"errors"
+	"log"
+	"sync"
+
+	"github.com/pion/webrtc/v3"
+	"maunium.net/go/mautrix/event"
+	"maunium.net/go/mautrix/id"
+)
+
+type LocalTrackInfo struct {
+	StreamID string
+	TrackID  string
+	Call     *Call
+}
+
+type LocalTrackWithInfo struct {
+	Track *webrtc.TrackLocalStaticRTP
+	Info  LocalTrackInfo
+}
```

Comment:
This now looks like an idea much worse than when I first thought of it... This could definitely be done differently now though I think it makes sense to just try to redo this during switching to using a pool of streams since that might change the design again...